File: LeetCodeProblemsCMake/Problems/LeetCode3335TotalCharactersInStringAfterTransformationsI.cpp
//
// Created by saeed on 5/13/25.
//

#include "LeetCode3335TotalCharactersInStringAfterTransformationsI.h"

#include <unordered_map>
#include <vector>
using namespace  std;
int LeetCode3335TotalCharactersInStringAfterTransformationsI::lengthAfterTransformations(string s, int t) {
    vector<long long> frequencies(26,0);
    for(int i = 0; i<s.length(); i++) {
        frequencies[s[i]-'a']++;
    }
    long long temp = 0;
    long long temp2 = 0;
    long long modulo=1'000'000'007;
    for(int i = 0; i<t; i++) {
        temp=0;
        temp2=0;
        for(int j = 0; j<25; j++) {
            temp=frequencies[j];
            frequencies[j] = temp2;
            temp2 = temp;
        }
        frequencies[0]+=frequencies[25]%modulo;
        frequencies[0]%=modulo;
        frequencies[1]+=frequencies[25]%modulo;
        frequencies[1]%=modulo;
        frequencies[25]=temp;
    }
    long long result = 0;
    for(int i = 0; i<26; i++) {
        result+= (frequencies[i]%modulo);
        result %= modulo;
    }
    return result;
}

File: LeetCodeProblemsGo/LeetCode3335TotalCharactersInStringAfterTransformationsI/LeetCode3335TotalCharactersInStringAfterTransformationsI.go
package main

func lengthAfterTransformations(s string, t int) int {
	frequencies := make([]int64, 26)
	for _, char := range s {
		frequencies[char-'a']++
	}
	modulo := int64(1e9 + 7)
	var temp int64
	var temp2 int64
	for i := 0; i < t; i++ {
		temp = 0
		temp2 = 0
		for j := 0; j < 25; j++ {
			temp = frequencies[j] % modulo
			frequencies[j] = temp2
			temp2 = temp
		}

		frequencies[0] += frequencies[25] % modulo
		frequencies[0] %= modulo
		frequencies[1] += frequencies[25] % modulo
		frequencies[1] %= modulo
		frequencies[25] = temp
	}
	var result int64 = 0
	for i := 0; i < 26; i++ {
		result += frequencies[i] % modulo
		result %= modulo
	}
	return int(result)
}
